PLMS/Controllers/TaskController.cs

```diff
@@ -1,115 +1,159 @@
+            if (!ModelState.IsValid)
+            {
+                return ApiResponseHelper.CreateErrorResponse(ModelState);
+            }
+            var taskDto = _mapper.Map<AddTaskDto>(model);
+            await _taskService.AddTaskAsync(taskDto, userId);
+            var createdTaskModel = _mapper.Map<AddTaskModel>(taskDto);
 
-            return CreatedAtAction(nameof(GetByIdAsync), new { id = task.Id }, task);
+            return ApiResponseHelper.CreateResponse(true, "Task was added successfully");
+            // CreatedAtAction(nameof(GetByIdAsync), new { id = taskDto.Id }, createdTaskModel);  -- Не получается вернуть маршрут на новое айди, не могу понять в чем ошибка (System.InvalidOperationException: No route matches the supplied values.
+         //   at Microsoft.AspNetCore.Mvc.CreatedAtActionResult.OnFormatting(ActionContext context)
         }
 
         [HttpDelete]
         public async Task<ActionResult> DeleteAsync(int id)
         {
-            var task = await _taskService.GetTaskByIdAsync(id);
+            string userId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            GetTaskDto taskDto;
+            try
+            {
+                taskDto = await _taskService.GetTaskByIdAsync(id, userId);
+            }
+            catch (UnauthorizedAccessException ex)
+            {
+                return Unauthorized(ex.Message);
+            }
 
-            if (task == null)
+            if (taskDto == null)
             {
-                return NotFound();//ToDo: should be success, see idempotent requests
+                return ApiResponseHelper.CreateResponse(true, "Task was deleted successfully");
             }
 
             await _taskService.DeleteTaskAsync(id);
 
-            return NoContent();//ToDo: better to just return 200
+            return ApiResponseHelper.CreateResponse(true, "Task was deleted successfully");
         }
 
         [HttpGet("GetFilteredShort")]
-        public async Task<ActionResult<IEnumerable<TaskDto>>> GetFilteredShortAsync([FromQuery] MyItemFilter filters,
-            [FromQuery] string sortField, [FromQuery] string includeColumns)
-            //ToDo: remove sortField - it's already present in the filters object;
-            //ToDo: remove includeColumns - they must not be exposed to the outside!
+        public async Task<ActionResult<IEnumerable<TaskShortModel>>> GetFilteredShortAsync([FromQuery] TaskItemFilter filters)
         {
-            //filters.UserName = base.User.Identity.Name; //ToDo: include user in the filters param;
-            //it should be always setup from the backend
-
-            var tasks = await _taskService.GetFilteredShortTasksAsync(filters, sortField, includeColumns);
+            filters.UserId = User.FindFirstValue(ClaimTypes.NameIdentifier);
+            var taskDto = await _taskService.GetFilteredShortTasksAsync(filters);
+            var taskModel = taskDto.Select(t => _mapper.Map<TaskShortModel>(t));
```

Comment:
You don't have to map items of enumarable objects one by one, AutoMapper is capable of doing that automatically.
You can try something like this:
_maper.Map<List<Task>, List<TaskShortModel>>(taskDto);